user-provided file template

virginiayung opened the pull request:
Adding the option for user-provided filename template with a default of `LC8.*\_B{b}.TIF`, where {b} is the band # to be extracted.

Ticket: https://github.com/mapbox/rio-toa/issues/9

virginiayung committed:
user-provided file template

virginiayung committed:
don't forget to update fixtures

perrygeo commented:
@virginiayung L 👍 TM

Should we also apply the same option to the `radiance` command?

perrygeo commented:
Also it would be good to include would be a direct unit test of the `_parse_bands_from_filename` with a few different templates: default, good and failing. Specifically it would be good to test/define the behavior when that function _can't_ find a band index from the given path and template.

virginiayung committed:
more tests

virginiayung committed:
more fail templates

virginiayung committed:
more _parse_bands_from_filename tests

virginiayung commented:
- [x] added `readtemplate` option for radiance
- [x] added unit tests for `_parse_bands_from_filename` with a few different templates: default, good and failing. 
- [x] added unit tests for reflectance cli and radiance cli
